ovh/resource_ovh_dns_record.go

```diff
@@ -0,0 +1,183 @@
+package ovh
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+type NewRecord struct {
+	Target    string `json:"target"`
+	Ttl       int    `json:"ttl"`
+	FieldType string `json:"fieldType"`
+	SubDomain string `json:"subDomain"`
+}
+
+type Record struct {
+	Id        int    `json:"id"`
+	Zone      string `json:"zone"`
+	Target    string `json:"target"`
+	Ttl       int    `json:"ttl"`
+	FieldType string `json:"fieldType"`
+	SubDomain string `json:"subDomain"`
+}
+
+func resourceOVHDomainZoneRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOVHRecordCreate,
+		Read:   resourceOVHRecordRead,
+		Update: resourceOVHRecordUpdate,
+		Delete: resourceOVHRecordDelete,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"target": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"ttl": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "3600",
+			},
+			"fieldType": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"subDomain": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceOVHRecordCreate(d *schema.ResourceData, meta interface{}) error {
+	provider := meta.(*Config)
+
+	// Create the new record
+	newRecord := &NewRecord{
+		FieldType: d.Get("fieldType").(string),
+		SubDomain: d.Get("subDomain").(string),
+		Target:    d.Get("target").(string),
+	}
+
+	newRecord.Ttl, _ = strconv.Atoi(d.Get("ttl").(string))
+
+	log.Printf("[DEBUG] OVH Record create configuration: %#v", newRecord)
+
+	resultRecord := Record{}
+
+	err := provider.OVHClient.Post(
+		fmt.Sprintf("/domain/zone/%s/record", d.Get("zone").(string)),
+		newRecord,
+		&resultRecord,
+	)
+
+	if err != nil {
+		return fmt.Errorf("Failed to create OVH Record: %s", err)
+	}
+
+	d.SetId(strconv.Itoa(resultRecord.Id))
+	d.Set("id", strconv.Itoa(resultRecord.Id))
+	log.Printf("[INFO] OVH Record ID: %s", d.Id())
+
+	return resourceOVHRecordRead(d, meta)
+}
+
+func resourceOVHRecordRead(d *schema.ResourceData, meta interface{}) error {
+	provider := meta.(*Config)
+
+	recordID, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error converting Record ID: %s", err)
+	}
+
+	record := Record{}
+	err = provider.OVHClient.Get(
+		fmt.Sprintf("/domain/zone/%s/record/%d", d.Get("zone").(string), recordID),
+		&record,
+	)
+
+	if err != nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("id", record.Id)
```

Comment:
`record.Id` is an `int`, can't use it for `d.Set()`. Need to use `fmt.Sprintf("%d", record.Id)` or `strconv`